src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/AttackSystem.java

```diff
@@ -22,66 +22,108 @@
-                CharacterHeldItemComponent characterHeldItemComponent = entity.getComponent(CharacterHeldItemComponent.class);
-                EntityRef heldItem = characterHeldItemComponent.selectedItem;
-                if (heldItem.getComponent(BlockItemComponent.class).blockFamily.getURI().toString().equals(LASUtils.BLACK_FLAG_URI)
-                        || heldItem.getComponent(BlockItemComponent.class).blockFamily.getURI().toString().equals(LASUtils.RED_FLAG_URI)) {
-                    Vector3f position = new Vector3f(attackingPlayer.getComponent(LocationComponent.class).getLocalPosition());
-                    Vector3f direction = localPlayer.getViewDirection();
-                    Vector3f newPosition = new Vector3f(position.x + direction.x * NEW_DIRECTION_MODIFIER,
-                            position.y + direction.y * NEW_DIRECTION_MODIFIER,
-                            position.z + direction.z * NEW_DIRECTION_MODIFIER
-                    );
-                    Vector3f impulseVector = new Vector3f(direction);
-                    entity.send(new DropItemRequest(heldItem, entity, impulseVector, newPosition));
+            if (entity.hasComponent(PlayerCharacterComponent.class)) {
+                blackFlagSlot = searchInventoryForFlag(entity, LASUtils.BLACK_FLAG_URI);
+                if (blackFlagSlot != EntityRef.NULL) {
+                    Vector3f position = new Vector3f(entity.getComponent(LocationComponent.class).getLocalPosition());
+                    Vector3f impulseVector = new Vector3f(attackingPlayer.getComponent(LocationComponent.class).getLocalPosition());
+                    entity.send(new DropItemRequest(blackFlagSlot, entity, impulseVector, position));
+                    return;
+                }
+                redFlagSlot = searchInventoryForFlag(entity, LASUtils.RED_FLAG_URI);
+                if (redFlagSlot != EntityRef.NULL) {
+                    Vector3f position = new Vector3f(entity.getComponent(LocationComponent.class).getLocalPosition());
+                    Vector3f impulseVector = new Vector3f(attackingPlayer.getComponent(LocationComponent.class).getLocalPosition());
+                    entity.send(new DropItemRequest(redFlagSlot, entity, impulseVector, position));
+                    return;
+                }
+            }
+        }
+    }
+
+    // Checks to see if player picks up flag of the same team
+    // If so, moves flag back to base
+    @ReceiveEvent
+    public void onInventorySlotChanged(InventorySlotChangedEvent event, EntityRef entity) {
+        EntityRef playerEntity = entity;
+        EntityRef item = event.getNewItem();
+        if (playerEntity.hasComponent(LASTeamComponent.class) && item.hasComponent(BlockItemComponent.class)) {
+            // If player team and flag team are the same, return flag to base
+            if (item.getComponent(BlockItemComponent.class).blockFamily.getURI().toString().equals(LASUtils.RED_FLAG_URI)
```

Comment:
We should stop doing URI matches for checks.
Let's create a `RedFlagComponent`. Define the flag prefabs to have this component by default. All we need to do is check if the item has this component or not from now.

Reply:
So I've had a problem from the beginning with finding a `Flag` or `BlackFlag` / `RedFlag` component attached to a flag block. In running `dumpEntities` I find an entity with the BlackFlagComponent attached:

```
{
      "id": 71,
      "parentPrefab": "engine:blockType",
      "engine:BlockType": {
        "block": "LightAndShadowResources:blackFlag"
      },
      "LightAndShadow:Flag": {},
      "LightAndShadow:LASTeam": {
        "team": "black"
      },
      "LightAndShadow:BlackFlag": {},
      "LightAndShadow:TakeBlockOnActivate": {}
    },
```

but in the inventory is a different entity reference: 
```
"entity": [
    {
      "id": 2321,
      "parentPrefab": "engine:blockItemBase",
      "owner": 132,
      "engine:DisplayName": {
        "name": "Blackflag"
      },
      "engine:BlockItem": {
        "blockFamily": "LightAndShadowResources:blackFlag"
      },
      "engine:BoxShape": {
        "extents": [
          0.92,
          0.92,
          0.92
        ]
      },
      "engine:Item": {
        "stackId": "block:LightAndShadowResources:blackFlag"
      },
      "engine:Mesh": {
        "translucent": false,
        "material": "engine:terrain",
        "color": [
          255,
          255,
          255,
          255
        ],
        "selfLuminance": 0.0,
        "hideFromOwner": false,
        "mesh": "engine:blockmesh#LightAndShadowResources:blackFlag"
      }
    },
```
This is why checking if the blockFamily String = URI works, but checking for the prefab does not. But I'm not sure why this is the case, and how to get a check for the prefab to work properly. Right now checking if the flag has a component, even when it does, fails.

I think this could be because it's a block that's placed into the world in the world render? But other components work, like the `TakeBlockOnActivate` check.